IdentityCore/src/util/NSURL+MSIDExtensions.m

```diff
@@ -33,15 +33,15 @@ @implementation NSURL (MSIDExtensions)
 // Decodes configuration contained in a URL fragment
 - (NSDictionary *)msidFragmentParameters
 {
-    return [NSDictionary msidDictionaryFromWWWFormURLEncodedString:self.fragment];
+    NSURLComponents* components = [NSURLComponents componentsWithURL:self resolvingAgainstBaseURL:YES];
+    return [NSDictionary msidDictionaryFromURLEncodedString:components.percentEncodedFragment];
```

Comment:
previous implementation didn't pass percentEncodedFragment, it passed just fragment. Just to double check that it won't cause issues.

Reply:
previously, for non standard URL like urn:ietf:wg:oauth:2.0:oob, any fragments and queries were dropped.